### PR TITLE
Adds CI Testing for Linux and Package Intallation

### PR DESCRIPTION
**Pull request by oke-aditya**
- As discussed. We continue support for Linux only, stick to PyTorch 1.5 only for now.

Packaging CI will check only on Ubuntu if we are able to build `.whl` of our repo.
It contains the entire installation Procedure that is required for end user and tries to Mimic it.

In other PR I will add PyPI releasing too.

Do let me know if changes are needed.

As a backtrace and reference for future. #226 Was the Orignal CI PR which was too extensive.

To make this into a single commit. Squash and Merge it !!

### CODE REVIEW

**Comment by oke-aditya** [comment]
Since Ross has fixed for PyTorch 1.6 with [this](https://github.com/rwightman/efficientdet-pytorch/commit/b0473e51bdda7cede6bc5d018e9b82ff93c792f4) commt. I'm giving one final try to it. If it works. We will pin to 1.6

**Comment by oke-aditya** [comment]
Bingo !!! WoW !!! All the tests passed on PyTorch 1.6. Seems people have fixed up. We can pin to PyTorch 1.6 for now. I see no trouble. 

This is ready to merge @lgvaz @ai-fast-track

**Comment by oke-aditya** [comment]
Helps #209 #182 #157 .

File: .github/workflows/mk-docs-build.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ on:
 jobs:
   build:
 
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
```
Comment (lgvaz) [inline .github/workflows/mk-docs-build.yml:10]: I believe it's best to pin an ubuntu version here, stick with 18.04

File: .github/workflows/mk-docs-deploy.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@ on:
 jobs:
   build:
 
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
```
Comment (lgvaz) [inline .github/workflows/mk-docs-deploy.yml:11]: same here

File: .github/workflows/mk-docs-build.yml (hunk 1 above)
Comment (oke-aditya) [reply .github/workflows/mk-docs-build.yml:10]: K I will Pin everything to Ubuntu 18.04 for now. We can bump it later.

File: .github/workflows/black.yml [hunk 3]
```diff
@@ -18,4 +18,4 @@ jobs:
     - name: Black Code Formatter
       uses: lgeiger/black-action@v1.0.1
       with:
-          args: "mantisshrimp --check"
+          args: ". --check"
```
[the remark this answers is not in the document]
Comment (oke-aditya) [reply .github/workflows/black.yml:21]: Changed it to mantis Folder

File: .github/workflows/mk-docs-deploy.yml (hunk 2 above)
Comment (oke-aditya) [reply .github/workflows/mk-docs-deploy.yml:11]: Pinned to Ubuntu 18.04

File: examples/training.py [hunk 4]
```diff
@@ -29,8 +29,12 @@
 valid_ds = Dataset(valid_records, datasets.pets.valid_albumentations_tfms_pets())
```
[the remark this answers is not in the document]
Comment (oke-aditya) [reply examples/training.py:29]: Restored the formatting

**Comment by oke-aditya** [comment]
If tests fail due to invalid SHA key error we can simply ignore them. It should work fine for 2/3 Python versions.
This issue is due to GitHub.

**Pull request merged**